docs/src/diplomacy/adder_tutorial.md

```diff
@@ -0,0 +1,322 @@
+# The Littlest Diplomacy Walkthrough
+
+- [What is Diplomacy?](#what-is-diplomacy)
+- [Diplomatic Adder and Test Bench](#diplomatic-adder-and-test-bench)
+  * [Parameter negotiation and passing](#parameter-negotiation-and-passing)
+    + [Parameters](#parameters)
+    + [Node Implementation](#node-implementation)
+    + [Nodes](#nodes)
+  * [Creating the `LazyModule`s](#creating-the-lazymodules)
+  * [Creating the Top](#creating-the-top)
+  * [The Generated Verilog](#the-generated-verilog)
+- [Other Resources](#other-resources)
+
+## What is Diplomacy?
+
+From the [Diplomatic Design Patterns: A TileLink Case Study](https://carrv.github.io/2017/papers/cook-diplomacy-carrv2017.pdf):
+> Diplomacy is a parameter negotiation framework for generating parameterized 
+  protocol implementations.
+
+The goal of this walkthrough is to demonstrate an extremely simple Diplomacy
+protocol. In this walkthrough, we will demonstrate how to create a parameterized
+adder and its associated testing modules.
+
+## Diplomatic Adder and Test Bench
+
+Let's begin by describing the desired circuit. We want a 2-to-1 **adder**, and
+to test it, we would like a **top-level test bench**, a **driver**, and a simple
+**monitor**.
+
+![adder test bench module hierarchy](diagrams/adder_modules.pdf)
+
+Let's say we would also like to parameterize the widths of these modules' ports.
+For the sake of demonstration, when negotiating parameters, our protocol will
+use the smaller of the two widths, which is the opposite of what typical Chisel
+width inference would do.
+
+```scala mdoc:invisible
+import chipsalliance.rocketchip.config.{Config, Parameters}
+import chisel3._
+import chisel3.internal.sourceinfo.SourceInfo
+import chisel3.stage.ChiselStage
+import chisel3.util.random.FibonacciLFSR
+import freechips.rocketchip.diplomacy.{SimpleNodeImp, RenderedEdge, ValName, SourceNode,
+                                       NexusNode, SinkNode, LazyModule, LazyModuleImp}
+```
+
+### Parameter negotiation and passing
+
+We have a width parameter we would like our modules to share. Any module that
+wants to send or receive parameter information must have **one or more nodes**.
+
+Between each pair of nodes, there may be **one or more edges**. The edge does
+the negotiation work to ensure parameter agreement between the nodes it connects.
```

Comment:
`Edges are directed, and if we draw their direction with an arrow we can then define the tail of the arrow as the **source** of the edge and the head of the arrow as the **sink** of the edge.`